tests/dialects/test_mysql.py

```diff
@@ -506,6 +506,33 @@ def test_mysql_time(self):
         self.validate_identity("FROM_UNIXTIME(a, b)")
         self.validate_identity("FROM_UNIXTIME(a, b, c)")
         self.validate_identity("TIME_STR_TO_UNIX(x)", "UNIX_TIMESTAMP(x)")
+        self.validate_all(
+            "SELECT TO_DAYS(x)",
+            write={
+                "mysql": "SELECT (DATEDIFF(DATE(x), DATE('0000-01-01')) + 1)",
+                "presto": "SELECT (DATE_DIFF('DAY', CAST(CAST('0000-01-01' AS TIMESTAMP) AS DATE), CAST(CAST(x AS TIMESTAMP) AS DATE)) + 1)",
+            },
+        )
+        self.validate_all(
+            "SELECT DATEDIFF(x, y)",
+            write={"mysql": "SELECT DATEDIFF(x, y)", "presto": "SELECT DATE_DIFF('day', y, x)"},
+        )
+        self.validate_all(
+            "DAYOFYEAR(x)",
+            write={"mysql": "DAYOFYEAR(DATE(x))"},
```

Comment:
if you wanted to make this identity, you could remove the date right?

Reply:
This is a side effect of representing `DAYOFYEAR(x)` using a `TsOrDsToDate` wrapper for `x` right?

Reply:
yea but in the gen of all these functions, we can check for tsords

Reply:
ok, I gave that a shot